extensions/TheShovel/ShovelUtilsTW.js

```diff
@@ -0,0 +1,257 @@
+          text: "Get brightness of [color]",
+          arguments: {
+            color: {
+              type: Scratch.ArgumentType.STRING,
+              defaultValue: '#ffffff',
+            }
+         }
+         },
+         
+         
+
+      ]
+      }
+    }
+
+
+    importImage({TEXT,NAME}) {
+      fetch(TEXT)
+  .then((r) => r.arrayBuffer())
+  .then((arrayBuffer) => {
+    const storage = vm.runtime.storage;
+    vm.addCostume(NAME+'.PNG', {
+      name: NAME,
+      asset: new storage.Asset(
+        storage.AssetType.ImageBitmap,
+        null, // asset id, doesn't need to be set here because of `true` at the end will make Scratch generate it for you
+        storage.DataFormat.PNG,
+        new Uint8Array(arrayBuffer),
+        true
+      )
+    })
+  });
+  }
+importSprite ({TEXT}) {
+  fetch(TEXT)
+  .then(r => r.arrayBuffer())
+  .then(buffer => vm.addSprite(buffer))
+  .then(() => {
+    console.log("Done");
+  })
+  .catch((error) => {
+    console.log("Error", error);
+  });
+}
+
+importSound ({TEXT,NAME}) {
+fetch(TEXT)
+  .then((r) => r.arrayBuffer())
+  .then((arrayBuffer) => {
+    const storage = vm.runtime.storage;
+    const asset = new storage.Asset(
+      storage.AssetType.Sound,
+      null,
+      storage.DataFormat.MP3,
+      new Uint8Array(arrayBuffer),
+      true
+    );
+    vm.addSound({
+      md5: NAME+'.mp3',
+      asset: asset,
+      name: NAME
+    });
+  });
+}
+
+importProject ({TEXT}) {
+  fetch(TEXT)
+  .then(r => r.arrayBuffer())
+.then(buffer => vm.loadProject(buffer))
+  .then(() => {
+    console.log("Done");
+    vm.greenFlag()
+  })
+  .catch((error) => {
+    console.log("Error", error);
+  });
+}
+
+restartProject(){
+vm.greenFlag()
+}
+
+loadExtension({TEXT}){
+  vm.extensionManager.loadExtensionURL(TEXT)
+}
+
+getlist({TEXT}){
+  return vm.runtime.getTargetForStage().lookupVariableByNameAndType(TEXT, 'list').value.toString()
+}
+setlist({TEXT,NAME}){
+  temp = JSON.parse(TEXT)
+  vm.runtime.getTargetForStage().lookupVariableByNameAndType(NAME, 'list').value = temp
+}
+
+setedtarget({NAME}){
+  vm.setEditingTarget(vm.runtime.getSpriteTargetByName(NAME).id); 
+}
+
+/**
+ * Calculate brightness value by RGB or HEX color.
+ * @param color (String) The color value in RGB or HEX (for example: #000000 || #000 || rgb(0,0,0) || rgba(0,0,0,0))
+ * @returns (Number) The brightness value (dark) 0 ... 255 (light)
+ */
+brightnessByColor ({color}) {
+  var color = "" + color, isHEX = color.indexOf("#") == 0, isRGB = color.indexOf("rgb") == 0;
+  if (isHEX) {
+    const hasFullSpec = color.length == 7;
+    var m = color.substr(1).match(hasFullSpec ? /(\S{2})/g : /(\S{1})/g);
+    if (m) var r = parseInt(m[0] + (hasFullSpec ? '' : m[0]), 16), g = parseInt(m[1] + (hasFullSpec ? '' : m[1]), 16), b = parseInt(m[2] + (hasFullSpec ? '' : m[2]), 16);
+  }
+  if (isRGB) {
+    var m = color.match(/(\d+){3}/g);
+    if (m) var r = m[0], g = m[1], b = m[2];
```

Comment:
```suggestion
    if (m) r = m[0], g = m[1], b = m[2];
```
fixes r is already defined